a10_neutron_lbaas/handler_base.py

```diff
@@ -13,15 +13,25 @@
 #    under the License.
 
 import json
+import logging
+
+LOG = logging.getLogger(__name__)
 
 
 class HandlerBase(object):
 
     def __init__(self, a10_driver):
         self.a10_driver = a10_driver
-        self.hooks = a10_driver.hooks
         self.openstack_driver = self.a10_driver.openstack_driver
 
+        # XXX(dougwig): is this too racy?
+        if self.a10_driver.provider is None:
+            for prov, driver in self.openstack_driver.plugin.drivers.items():
+                if driver is self.openstack_driver:
+                    self.a10_driver._late_init(prov)
```

Comment:
Why isn't this a10_openstack_lb's job to set it's provider if it's None? Something like

```
def __init__(..., provider=None):
    self._provider = provider

@property
def provider(self):
    """neutron lbaas provider name. The provider name is not available during __init__"""
    if self._provider is None:
        ...
            self._provider = ...
    return self._provider
```

This also gets ugly when we also need to initialize hooks based on config with also depends on the provider.